pkg/api/secretbootstrap/testdata/TestRoundtripConfig/basic_base.yaml

```diff
@@ -0,0 +1,58 @@
+cluster_groups:
```

Comment:
So notably we do not have overlapping groups here, right? If we want to make a statement that there should never be an item that contains the same cluster from more than one group, we should enforce that via validation

Reply:
I'm not sure I understand exactly what you mean here. I made the test config intentionally contain cases that are possible, but not currently within the actual data. I am fine with adding that validation and removing the case from the test config, but wasn't sure if we would _never_ want that.

Reply:
I don't see a config entry that contains both `build_farm` and `cg_2` or `build_farm` and `cg_3` - what entry currently overlaps?

Reply:
Oh I get what you are saying now! I'll try that out and, assuming it doesn't function properly, add some validation, thanks

Reply:
I modified the cluster_groups to create an overlap in `cg_2` and `cg_3`, and the test still passes so it seems like we are good here